cmd/attack/http.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attack
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"go.uber.org/fx"
+
+	"github.com/chaos-mesh/chaosd/cmd/server"
+	"github.com/chaos-mesh/chaosd/pkg/core"
+	"github.com/chaos-mesh/chaosd/pkg/server/chaosd"
+	"github.com/chaos-mesh/chaosd/pkg/utils"
+)
+
+func NewHTTPAttackCommand(uid *string) *cobra.Command {
+	option := core.NewHTTPAttackOption()
+	dep := fx.Options(
+		server.Module,
+		fx.Provide(func() *core.HTTPAttackOption {
+			option.UID = *uid
+			return option
+		}),
+	)
+
+	cmd := &cobra.Command{
+		Use:   "http <subcommand>",
+		Short: "HTTP attack related commands",
+	}
+
+	cmd.AddCommand(
+		NewHTTPAbortCommand(dep, option),
+		NewHTTPDelayCommand(dep, option),
+		NewHTTPFileCommand(dep, option),
+	)
+
+	return cmd
+}
+
+func NewHTTPAbortCommand(dep fx.Option, o *core.HTTPAttackOption) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "abort",
+		Short: "abort selected HTTP Package",
+		Run: func(*cobra.Command, []string) {
+			o.Action = core.HTTPAbortAction
+			abort := true
+			o.Rule.Actions.Abort = &abort
+			utils.FxNewAppWithoutLog(dep, fx.Invoke(processHTTPAttack)).Run()
+		},
+	}
+
+	setTarget(cmd, o)
+	setSelector(cmd, o)
+	return cmd
+}
+
+func NewHTTPDelayCommand(dep fx.Option, o *core.HTTPAttackOption) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delay",
+		Short: "delay selected HTTP Package",
+		Run: func(*cobra.Command, []string) {
+			o.Action = core.HTTPDelayAction
+			utils.FxNewAppWithoutLog(dep, fx.Invoke(processHTTPAttack)).Run()
+		},
+	}
+
+	setTarget(cmd, o)
+	setSelector(cmd, o)
+
+	delay := ""
+	o.Rule.Actions.Delay = &delay
+	cmd.Flags().StringVarP(o.Rule.Actions.Delay, "delay time", "d", "", "Delay represents the delay of the target request/response.")
+	return cmd
+}
+
+func setTarget(cmd *cobra.Command, o *core.HTTPAttackOption) {
+	cmd.Flags().UintSliceVarP(&o.ProxyPorts, "proxy_ports", "p", nil,
```

Comment:
how about changing to `proxy-ports` to keep the naming styles consistent

Reply:
OK

Reply:
Fixed.